### DISCUSSION TITLE
`autopaper.sty` template and documentation

### DISCUSSION THREAD

**Discussion by sgbaird** [sparks-baird/auto-paper#4] [Polls]

**Comment by jamesmrollins** [comment]
@sgbaird I am supporting a principal investigator in a national NIH funded study.  The study is resulting in periodic manuscript production that I would like to automate using GitHub actions and API's.  From my preliminary investigation of YouTube and Google, your efforts appear to stand alone.  I am looking to collaborate with you and have resources that I can apply to this effort.

See [mtl.how/demo](https://mtl.how/demo) for more information about the project.

**Reply by sgbaird** [reply]
Hi @jamesmrollins, thanks for your interest. You may be interested in [`show-your-work`](https://github.com/showyourwork/showyourwork) https://github.com/sparks-baird/auto-paper/issues/6, especially if you're trying to automate the production of the manuscript using GitHub actions. At the very least, some of the tools that `show-your-work` uses would probably be of help.

- How frequently will the manuscript be put into production?
- What kind of changes will be made? (major vs. minor, modifying the document structure vs. replacing images and modifying text)
- Will this be used for multiple manuscripts?
- Any other requirements?

Thanks for including the link to the project.

**Reply by sgbaird** [reply]
You may also want to have a look at the discussion here about Quarto https://github.com/sparks-baird/auto-paper/discussions/7

Are you planning on having the manuscript source files in LaTeX, Markdown, RST, or something else?

See also https://pandoc.org/

**Comment by jamesmrollins** [comment]
Hi @sgbaird - see below for responses to your questions:

- We are looking publish 5-7 papers this year.
- I would like to set up an automated structure that makes it possible to merge text, tables, r-code, r-charts, tables and Zotero bibliography. 
- Manuscripts are managed in Open Science Framework, so able to support markdown outputs and it's supported by an API. 
- I think LaTex offers the most robust typesetting controls.
- It would be cool if we could toggle between formats using some form of CSS between, say, an APA output and a Chicago Output depending on the publisher's requirements. 

Also, I should mention that I am not super technical.  I am the "use case" and "wireframe" guy.  I do have a Dev on my team that I can resource and prioritize.

**Reply by sgbaird** [reply]
Thanks for the extra info! Lmk where you think I can help, and I'll see what's possible for me.

**Comment by jamesmrollins** [comment]
Sorry for the delayed response @sgbaird - I had to break for medical leave last week.  Just to confirm, your action basically takes all the inputs from various sources (GitHub, OSF, Zotero) and can merge it into a LaTeX standard document, correct?  If so, then we will review your code and set up some use cases.

**Reply by sgbaird** [reply]
Are you referring to the `show-your-work` action? I didn't write this GitHub action, and I don't have one for this repo. `show-your-work` seems like a good way to go for me. The workflow shown in this repo works well for using Overleaf's integration with GitHub (unfortunately, now a paid feature of Overleaf for accounts post-dating 2018 IIRC). Not too familiar with OSF, so I can't speak to the integration there - but as long as you're just transferring files that GitHub can handle I don't see any issue. Zotero has some nice automatic export features, so I usually export these to a Google Drive, get the direct download link, and then add the file via the direct download link on Overleaf. You could also use a LaTeX typesetter on your local machine - I like to use TeXStudio. Once the initial setup is done, it's a bit simpler to make updates to documents that way - but you might run into issues where you have to figure out some bug that Overleaf has already taken care of (including doing custom setup of how the TeX commands are run and in what order). If the people making the documents are not too familiar with LaTeX, it's probably better to stick with Overleaf.
